### PR TITLE
new appServicePlan in webapponlinux

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java [hunk 1]
```diff
@@ -74,20 +81,31 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
...
+                    if (basePath == null) {
+                        errorln("Project base path is null.");
+                        throw new FileNotFoundException("Project base path is null.");
+                    }
                     // locate war file to specified location
                     println("Locate war file ...  ");
                     List<MavenProject> mavenProjects = MavenProjectsManager.getInstance(project).getRootProjects();
-                    String targetBuildPath = new File(mavenProjects.get(0).getBuildDirectory()).getPath()
-                            + File.separator + mavenProjects.get(0).getFinalName() + ".war";
+                    String targetBuildPath = new File(mavenProjects.get(0).getBuildDirectory()).getPath();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java]: null pointer of mavenProjects

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java [hunk 2]
```diff
@@ -97,22 +115,40 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
...
+                                .updateWebAppOnLinux(deployModel.getSubscriptionId(), deployModel.getWebAppId(),
+                                        acrInfo);
+                        if (app != null && app.name() != null) {
+                            println(String.format("URL:  http://%s.azurewebsites.net/%s", app.name(),
+                                    project.getName()));
+                        }
                     }
                     return null;
                 }
         ).subscribeOn(Schedulers.io()).subscribe(
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java]: use provider scheduler

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java [hunk 3]
```diff
@@ -192,6 +213,22 @@ public WebApp updateWebAppOnLinux(String sid, String webAppId, ImageSetting imag
...
+    /**
+     * List Web App on Linux in all selected subscriptions.
+     *
+     * @param force flag indicating whether force to fetch most updated data from server
+     * @return list of Web App on Linux (SiteInner instances)
+     */
+    public List<ResourceEx<SiteInner>> listAllWebAppsOnLinux(boolean force) {
+        List<ResourceEx<SiteInner>> ret = new ArrayList<>();
+        for (Subscription sb : AzureMvpModel.getInstance().getSelectedSubscriptions()) {
+            List<ResourceEx<SiteInner>> wal = AzureWebAppMvpModel.getInstance()
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java]: remove getinstance

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java (hunk 2 above)
Comment (Eskibear) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java]: cannot access, currently it's defined in MvpPresenter.
FYI @jdneo .

Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java]: We can directly get the provider from SchedulerProviderFactory: https://github.com/Microsoft/azure-tools-for-java/blob/develop/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebDeployUtil.java#L199

**Pull request merged**